### PR TITLE
parseHeaders flag support

### PR DESCRIPTION
**Pull request by Mikhail-Pranovich**

### CODE REVIEW

File: src/Microsoft.AspNet.TelemetryCorrelation/TelemetryCorrelationHttpModule.cs [hunk 1]
```diff
@@ -21,6 +21,11 @@ static TelemetryCorrelationHttpModule()
             onStepMethodInfo = typeof(HttpApplication).GetMethod("OnExecuteRequestStep");
         }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether TelemetryCorrelationHttpModule should parse headers to get correlation ids.
+        /// </summary>
+        public bool ParseHeaders { get; set; } = true;
```
Comment (lmolkova) [inline src/Microsoft.AspNet.TelemetryCorrelation/TelemetryCorrelationHttpModule.cs]: HttpModule is created per-request (and may be reused in some cases). Having it as a property on the module will require to set it every time. 

So, please use env var (which is cached somewhere outside of HttpModule or static) or make this property static.

File: test/Microsoft.AspNet.TelemetryCorrelation.Tests/ActivityHelperTest.cs [hunk 2]
```diff
@@ -259,7 +259,7 @@ public void Stop_Root_Activity_With_129_Nesting_Depth()
         public void Should_Not_Create_RootActivity_If_AspNetListener_Not_Enabled()
         {
             var context = HttpContextHelper.GetFakeHttpContext();
-            var rootActivity = ActivityHelper.CreateRootActivity(context);
```
Comment (lmolkova) [inline test/Microsoft.AspNet.TelemetryCorrelation.Tests/ActivityHelperTest.cs]: how about testing a case when headers are not parsed?

File: src/Microsoft.AspNet.TelemetryCorrelation/TelemetryCorrelationHttpModule.cs (hunk 1 above)
Comment (Mikhail-Pranovich) [reply src/Microsoft.AspNet.TelemetryCorrelation/TelemetryCorrelationHttpModule.cs]: I do have logic which controls the creation, and hence this logic is correct. Don't want to use envvars for this here.

File: test/Microsoft.AspNet.TelemetryCorrelation.Tests/ActivityHelperTest.cs (hunk 2 above)
Comment (Mikhail-Pranovich) [reply test/Microsoft.AspNet.TelemetryCorrelation.Tests/ActivityHelperTest.cs]: sure, will add the testcasse

File: src/Microsoft.AspNet.TelemetryCorrelation/TelemetryCorrelationHttpModule.cs (hunk 1 above)
Comment (lmolkova) [reply src/Microsoft.AspNet.TelemetryCorrelation/TelemetryCorrelationHttpModule.cs]: Also this should be less discoverable - if you could make it EditorBrowseable.Never - please do

Comment (lmolkova) [reply src/Microsoft.AspNet.TelemetryCorrelation/TelemetryCorrelationHttpModule.cs]: > Don't want to use envvars for this here.
then make this property static.

File: test/Microsoft.AspNet.TelemetryCorrelation.Tests/ActivityHelperTest.cs [hunk 3]
```diff
@@ -304,6 +304,24 @@ public void Can_Create_RootActivity_And_Restore_Info_From_Request_Header()
             Assert.Equal(expectedBaggage, actualBaggage);
         }
 
+        [Fact]
+        public void Can_Create_RootActivity_And_Ignore_Info_From_Request_Header_If_ParseHeaders_Is_False()
+        {
+            var requestHeaders = new Dictionary<string, string>
+            {
+                { ActivityExtensions.RequestIDHeaderName, "|aba2f1e978b2cab6.1" },
```
Comment (lmolkova) [inline test/Microsoft.AspNet.TelemetryCorrelation.Tests/ActivityHelperTest.cs]: this is not valid Id, please use `|aba2f1e978b2cab6.1.`

Comment (Mikhail-Pranovich) [reply test/Microsoft.AspNet.TelemetryCorrelation.Tests/ActivityHelperTest.cs]: this is taken from previous UT, should it be updated there as well?

Comment (lmolkova) [reply test/Microsoft.AspNet.TelemetryCorrelation.Tests/ActivityHelperTest.cs]: please update if you don't mind. Or just fix your test case. This is minor, does not affect result, at least today.

File: src/Microsoft.AspNet.TelemetryCorrelation/TelemetryCorrelationHttpModule.cs [hunk 4]
```diff
@@ -21,6 +22,18 @@ static TelemetryCorrelationHttpModule()
...
+        /// Gets or sets a value indicating whether TelemetryCorrelationHttpModule should parse headers by default to get correlation ids.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static bool ParseHeadersDefault { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether TelemetryCorrelationHttpModule should parse headers to get correlation ids.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool? ParseHeaders { get; set; } = null;
```
Comment (lmolkova) [inline src/Microsoft.AspNet.TelemetryCorrelation/TelemetryCorrelationHttpModule.cs]:
```suggestion
        public bool ParseHeaders { get; set; } = ParseHeadersDefault;
```

then there is no need in `ParseHeaders ?? ParseHeadersDefault`

Comment (Mikhail-Pranovich) [reply src/Microsoft.AspNet.TelemetryCorrelation/TelemetryCorrelationHttpModule.cs]: reverted the static variable for initial value.

**Comment by batkaevruslan** [comment]
> [Please make ParseHeader is static](https://github.com/aspnet/Microsoft.AspNet.TelemetryCorrelation/pull/47#pullrequestreview-176993785)
> Overall, this change is needed to save some performance in the cases where we don't care about cross-service correlation and this is a hook to prevent heavy header access. 

@lmolkova, @Mikhail-msft, hi! ParseHeaders property of TelemetryCorrelationHttpModule.cs is not static. Can I disable headers parsing?

**Pull request merged**